src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractWalkToProxy.java

```diff
@@ -0,0 +1,310 @@
+package com.minecolonies.coremod.entity.pathfinding;
+
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.EntityUtils;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Abstract class for the walkToProxy.
+ */
+public abstract class AbstractWalkToProxy
```

Comment:
At least define an Interface for this class in the API.
If it is possible move it completly to the API package as well next to the creation of the Interface.